### PR TITLE
[docs] - [definitions] Update OSS deployment overview

### PR DESCRIPTION
**Pull request by erinkcochran87**
This PR updates the OSS deployment overview for the new `Definitions` world. [Preview](https://dagster-git-erin-definitions-deploy-overview-elementl.vercel.app/deployment/overview).

It also:

- Edits copy for clarity / style
- Cleans up some formatting
- Adds `Code location servers` as a new term (replacing user repository deployment)
- Updates the `Workspace` copy to the more general `workspace files`

### CODE REVIEW

File: docs/content/deployment/overview.mdx [hunk 1]
```diff
@@ -13,52 +24,254 @@ width={2986}
...
+    <tr>
+      <td>
+        <a href="/concepts/code-locations">Code location server</a>
+      </td>
+      <td>
+        Code lcation servers serve metadata about the collection of its Dagster
+        definitions. You can have many code location servers; each server can
+        have one or more replicas.
+      </td>
+      <td>Supported</td>
```
Comment (erinkcochran87) [inline docs/content/deployment/overview.mdx:88]: Addition of `code location servers`

File: docs/content/deployment/overview.mdx [hunk 2]
```diff
@@ -13,52 +24,254 @@ width={2986}
...
+        <a href="/concepts/code-locations/workspace-files">Workspace</a>
+      </td>
+      <td>
+        <code>workspace.yaml</code>
+      </td>
+      <td>
+        Workspace files define how to access and load your code. You can define
+        workspace configuration using <code>workspace.yaml</code>.
+      </td>
+    </tr>
```
Comment (erinkcochran87) [inline docs/content/deployment/overview.mdx:157]: Changes related to Workspaces

File: docs/content/deployment/overview.mdx [hunk 3]
```diff
@@ -13,52 +24,254 @@ width={2986}
...
-| [Dagit](/concepts/dagit/dagit)                                                | Dagit serves the user interface and responds to GraphQL queries. It can have 1 or more replicas.                                               | Yes                                                                                |
-| [Dagster Daemon](/deployment/dagster-daemon)                                  | The Dagster Daemon operates schedules, sensors, and run queuing. Currently, replicas are not supported.                                        | Required for schedules, sensors, and run queuing                                   |
-| [User Repository Deployments](/concepts/repositories-workspaces/repositories) | User Repository Deployments serve metadata about user jobs. You can have many repository servers, and each server can have 1 or more replicas. | Recommended. Otherwise, Dagit and the Daemon must have direct access to user code. |
+**For larger deployments**, you can run and manage the code location servers separately.
```
Comment (rexledesma) [inline docs/content/deployment/overview.mdx:95]: Let's remove both of these comments.

This is a vestige of the fact that one of the paths for deploying Dagster was to have the Dagster definitions to be co-located in the server that hosts Dagit. We don't really recommend this path other than if you're developing locally.

File: docs/content/deployment/overview.mdx [hunk 4]
```diff
@@ -13,52 +24,254 @@ width={2986}
...
+        Currently, replicas are not supported.
+      </td>
+      <td>Not supported</td>
+    </tr>
+    <tr>
+      <td>
+        <a href="/concepts/code-locations">Code location server</a>
+      </td>
+      <td>
+        Code lcation servers serve metadata about the collection of its Dagster
```
Comment (rexledesma) [inline docs/content/deployment/overview.mdx:84]:
```suggestion
        Code location servers serve metadata about the collection of its Dagster
```

File: docs/content/deployment/overview.mdx [hunk 5]
```diff
@@ -13,52 +24,254 @@ width={2986}
...
+**For larger deployments**, you can run and manage the code location servers separately.
 
-### Configuring Dagster Deployments
+---
 
-Dagster deployments are composed of multiple components, such as storages, executors, and run launchers. One of the core features of Dagster is that each of these components is swappable and configurable.
+## Deployment configuration
 
-If you don't provide any custom configuration, Dagster automatically uses a default implementation of each component. However, you can swap out the default implementation for another Dagster-provided one or even write a custom implementation. For example, Dagster uses the <PyObject module="dagster._core.storage.runs" object="SqliteRunStorage" /> by default to store information about pipeline runs. You can swap it out with the Dagster-provided <PyObject module="dagster_postgres" object="PostgresRunStorage"/> instead or even write your own custom storage class.
+Dagster OSS deployments are composed of multiple components, such as storages, executors, and run launchers. One of the core features of Dagster is that each of these components is swappable and configurable.
```
Comment (rexledesma) [inline docs/content/deployment/overview.mdx:101]:
```suggestion
Dagster OSS deployments are composed of multiple components, such as storages, executors, and run launchers. One of the core features of Dagster is that each of these components is swappable and configurable.
```

File: docs/content/deployment/overview.mdx [hunk 6]
```diff
@@ -13,52 +24,254 @@ width={2986}
...
 
-Based on the component's scope, the component is configured either at the Dagster Instance or Job Run level. Access to user code is configured at the Workspace level. The table below describes how to configure components at each of these levels.
+If custom configuration isn't provided, Dagster will automatically use a default implementation of each component. However, default implementations can be swapped for another built-in Dagster option or your own custom implementation. For example, by default Dagster uses <PyObject module="dagster._core.storage.runs" object="SqliteRunStorage" /> to store information about pipeline runs. This can be swapped with the Dagster-provided <PyObject module="dagster_postgres" object="PostgresRunStorage"/> instead or or a custom storage class.
```
Comment (rexledesma) [inline docs/content/deployment/overview.mdx:103]: The last sentence from the previous paragraph can be merged with this paragraph. And then, this sentence can be deleted: "However, default implementations can be swapped for another built-in Dagster option or your own custom implementation"

**Review by rexledesma — APPROVED** [review]
Looks fine, just see the comments before merging!

**Pull request merged**